File: Ahorn/entities/JellyRefill.jl
module AnonhelperJellyRefill

using ..Ahorn, Maple

@mapdef Entity "Anonhelper/JellyRefill" JellyRefill(x::Integer, y::Integer, oneUse::Bool=false)


const placements = Ahorn.PlacementDict(
    "JellyRefill (Anonhelper)" => Ahorn.EntityPlacement(
        JellyRefill, 
	"point"
    )
)

sprite = "objects/AnonHelper/jellyRefill/idle00"

function getSprite(entity::JellyRefill)

    return sprite
end

function Ahorn.selection(entity::JellyRefill)
    oneUse = get(entity.data, "oneUse", false)
    x, y = Ahorn.position(entity)
    sprite = getSprite(entity)

    return Ahorn.getSpriteRectangle(sprite, x, y)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::JellyRefill, room::Maple.Room)
    sprite = getSprite(entity)
    Ahorn.drawSprite(ctx, sprite, 0, 0)
end

end

File: Loenn/entities/jelly_refill.lua
local utils = require("utils")

local jellyRefill = {}

jellyRefill.name = "Anonhelper/JellyRefill"
jellyRefill.depth = -100
jellyRefill.placements = {
    {
        name = "jelly_refill",
        data = {
            oneUse = false,
        }
    }
}
jellyRefill.texture = "objects/AnonHelper/jellyRefill/idle00"

function jellyRefill.selection(room, entity)
	local x, y = entity.x or 0, entity.y or 0
    return utils.rectangle(x - 5, y - 4, 10, 9)
end

return jellyRefill